tests/DataValues/TimeValueCalculatorTest.php

```diff
@@ -209,4 +209,279 @@ public function testGetSecondsForPrecision( $precision, $expected ) {
+			0, 0, 0,
+			TimeValue::PRECISION_DAY,
+			TimeValue::CALENDAR_GREGORIAN
+		);
+		$unixHigherTimestamp = $timeValueCalculator->getHigherTimestamp( $timeValue );
+		$this->assertEquals( -30530044801, $unixHigherTimestamp );
+		foreach ( [ '+', '-' ] as $sign ) {
+			$oldUnixHigherTimestampMonth = null;
+			foreach ( $timestamps as $timestamp ) {
+				$timestamp = $sign . $timestamp;
+				$timeValue = new TimeValue(
+					$timestamp,
+					0, 0, 0,
+					TimeValue::PRECISION_MONTH,
+					TimeValue::CALENDAR_GREGORIAN
+				);
+				$unixHigherTimestamp = $timeValueCalculator->getHigherTimestamp( $timeValue );
+				if ( !empty( $oldUnixHigherTimestampMonth ) ) {
+					if ( $sign === '+' ) {
+						$this->assertGreaterThan( $oldUnixHigherTimestampMonth, $unixHigherTimestamp );
+					} else {
+						$this->assertGreaterThan( $unixHigherTimestamp, $oldUnixHigherTimestampMonth );
+					}
+				}
+				$oldUnixHigherTimestampMonth = $unixHigherTimestamp;
+				$oldUnixHigherTimestamp = null;
+				foreach ( $precisions as $precision ) {
+					$timeValue = new TimeValue(
+						$timestamp,
+						0, 0, 0,
+						$precision,
+						TimeValue::CALENDAR_GREGORIAN
+					);
+					$unixTimestampAsIs = $timeValueCalculator->getTimestamp( $timeValue );
+					$unixHigherTimestamp = $timeValueCalculator->getHigherTimestamp( $timeValue );
+					$this->assertLessThanOrEqual( $unixHigherTimestamp, $unixTimestampAsIs );
+					$timeValueAfter1 = new TimeValue(
+						$timestamp,
+						0, 1, 1,
+						$precision,
+						TimeValue::CALENDAR_GREGORIAN
+					);
+					$unixHigherTimestampAfter1 = $timeValueCalculator->getHigherTimestamp( $timeValueAfter1 );
+					if ( $unixTimestampAsIs > -10000000000000 && $unixTimestampAsIs < 10000000000000 ) {
+						$this->assertLessThan( $unixHigherTimestampAfter1, $unixHigherTimestamp );
+					} else {
+						$this->assertLessThanOrEqual( $unixHigherTimestampAfter1, $unixHigherTimestamp );
+					}
+					$timeValueAfter2 = new TimeValue(
+						$timestamp,
+						0, 2, 2,
+						$precision,
+						TimeValue::CALENDAR_GREGORIAN
+					);
+					$unixHigherTimestampAfter2 = $timeValueCalculator->getHigherTimestamp( $timeValueAfter2 );
+					if ( $unixTimestampAsIs > -10000000000000 && $unixTimestampAsIs < 10000000000000 ) {
+						$this->assertLessThan( $unixHigherTimestampAfter2, $unixHigherTimestampAfter1 );
+					} else {
+						$this->assertLessThanOrEqual( $unixHigherTimestampAfter2, $unixHigherTimestampAfter1 );
+					}
+					// As $precisions are ordered from most to least precise, old higher timestamps must be
+					// less than or equal to current higher timestamps
+					if ( !empty( $oldUnixHigherTimestamp ) ) {
+						$this->assertLessThanOrEqual( $unixHigherTimestamp, $oldUnixHigherTimestamp );
+					}
+					$oldUnixHigherTimestamp = $unixHigherTimestamp;
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param string $timestamp
+	 * @param int $precision
+	 */
+	private function auxTestGetHigherTimestamp( $timestamp, $precision ) {
```

Comment:
Is this function used anywhere? I can’t find any uses of it in this pull request using in-browser search.

Reply:
No, sorry, I should have removed it.